gc/base/ParallelDispatcher.hpp

```diff
@@ -92,6 +92,52 @@ class MM_ParallelDispatcher : public MM_BaseVirtual
 	 * Function members
 	 */
 private:
+#if defined(J9VM_OPT_CRIU_SUPPORT)
+
+	/**
+	 * Expand/fill the thread pool by starting up threads based on what H/W supports.
+	 * This API is capped by the initial thread pool size, i.e expanding
+	 * past _poolMaxCapacity is not possible
+	 * (expanding the dispatcher tables is currently not supported).
+	 *
+	 * This API assumes CRIU is the only consumer (not tested for general use).
+	 * The following conditions are required while expanding the thread pool:
+	 *     1) Caller is NOT holding exclusive VM access.
+	 *     2) Dispatcher is idle (no task can be dispatched).
+	 *     3) Dispatcher can't be in/enter shutdown.
+	 *
+	 * @param[in] env the current environment.
+	 * @return boolean indicating if threads started up successfully.
+	 */
+	virtual bool expandThreadPool(MM_EnvironmentBase *env);
+
+	/**
+	 * Contract the thread pool by shutting down threads in the pool to obtain newThreadCount.
+	 *
+	 * This API assumes  CRIU is the only consumer (not tested for general use).
+	 * The following conditions are assumed while contracting the thread pool:
+	 *     1) API can only be called once during VM lifetime.
+	 *     2) Caller is NOT holding exclusive VM access.
+	 *     3) Dispatcher is idle (no task can be dispatched).
+	 *     4) Another party can't enter Dispatcher shutdown.
+	 *
+	 * @param[in] env the current environment.
+	 * @param[in] newThreadCount the number of threads to keep in the thread pool.
+	 * @return void
+	 */
+	virtual void contractThreadPool(MM_EnvironmentBase *env, uintptr_t newThreadCount);
```

Comment:
minor, but these methods don't really need to be virtual. in fact, even true for most of legacy methods (which perhaps made sense back at the time when we also had a base non parallel Dispatcher.

again, up to you if you want to simplify or not

Reply:
Cleaned up these + legacy methods by checking what is really overridden in Scheduler